lambda/utils.js

```diff
@@ -383,13 +383,8 @@ module.exports._fetchS3Object = async(bucket, key) => {
  * Generate a list of `num` payloads (repeated or weighted)
  */
 module.exports.generatePayloads = (num, payloadInput) => {
-    if (Array.isArray(payloadInput)) {
-        // if array, generate a list of payloads based on weights
-
-        // fail if empty list or missing weight/payload
-        if (payloadInput.length === 0 || payloadInput.some(p => !p.weight || !p.payload)) {
-            throw new Error('Invalid weighted payload structure');
-        }
+    if (Array.isArray(payloadInput) && utils.isWeightedPayload(payloadInput)) {
```

Comment:
We can drop the `Array.isArray(payloadInput)` check here 👌

Reply:
Done